compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1448,12 +1448,11 @@ impl<'tcx> InferCtxt<'tcx> {
         value.fold_with(&mut r)
     }
 
-    /// Returns the first unresolved variable contained in `T`. In the
-    /// process of visiting `T`, this will resolve (where possible)
-    /// type variables in `T`, but it never constructs the final,
-    /// resolved type, so it's more efficient than
-    /// `resolve_vars_if_possible()`.
-    pub fn unresolved_type_vars<T>(&self, value: &T) -> Option<(Ty<'tcx>, Option<Span>)>
+    /// Returns the first unresolved type or const variable contained in `T`.
+    /// In the process of visiting `T`, this will resolve (where possible)
+    /// type variables in `T`, but it never constructs the final, resolved
+    /// type, so it's more efficient than `resolve_vars_if_possible()`.
+    pub fn unresolved_non_region_vars<T>(&self, value: &T) -> Option<(ty::Term<'tcx>, Option<Span>)>
```

Comment:
With #101900, I think this should be called `unresolved_type_or_const_vars`

Reply:
Sure, though I would assume that effect vars, if they existed, would also need to be included in this, or else we'd have the same ICE